### DISCUSSION TITLE
Folder structure of Vid4

### DISCUSSION THREAD

**Discussion by hiroshi-mori-1998** [open-mmlab/mmediting#1264] [Datasets] [answered]
Hi, thank you for releasing the code of BasicVSR++ !
I want to test BasicVSR++ in Vid4, but I don't know the folder structure of it .
Could you tell me how to do it ?
Thanks !

**Comment by hiroshi-mori-1998** [comment]
In addition, it would be great if you could also share the configuration file for testing with Vid4.

**Comment by ckkelvinchan** [comment, marked as answer]
Thank you for your interests in MMEditing and BasicVSR++. 

1. The folder structure for Vid4 is as follow:
```
| GT
    | calendar
        | 00000000.png
        | 00000001.png
        | ...
    | foliage
        | ...
    | ...
| LQ
    | calendar
        | ...
    | .... 
 ```
I will put it in the documentation. 
2. For the configuration file, you can check `val_pipeline` and `data['val']` [here](https://github.com/open-mmlab/mmediting/blob/master/configs/restorers/basicvsr_plusplus/basicvsr_plusplus_c64n7_8x1_300k_vimeo90k_bi.py). We use Vid4 for validation and Vimeo-90K for test. If you want to use Vid4 for test, you can copy the validation settings to the test settings.

**Comment by dzz416** [comment]
if i want to use Vid4dataset as the test data,i will need the GT.txt,whats the structure of Vid4 txt?

**Comment by ckkelvinchan** [comment]
When you use `SRFolderMultipleGTDataset`, you do not need the annotation file.

**Comment by dzz416** [comment]
and if i want to use Vid4dataset as the val data,whats the structure of Vid4 txt?

**Comment by ckkelvinchan** [comment]
`SRVid4Dataset`is for the sliding-window framework such as EDVR. For recurrent framework such as BasicVSR and BasicVSR++, please use `SRFolderMultipleGTDataset`. For more details, please see [here](https://github.com/open-mmlab/mmediting/blob/master/configs/restorers/basicvsr_plusplus/basicvsr_plusplus_c64n7_8x1_300k_vimeo90k_bi.py#L20).

**Comment by ckkelvinchan** [comment]
Closing now due to inactivity. Please reopen if the problem persists.
